### ISSUE TITLE
T5ForConditionalGeneration fp16 nan loss

### ISSUE THREAD

**Issue by xwhan** [huggingface/transformers#5300]
# 🐛 Bug

## Information

Model I am using (Bert, XLNet ...):

Language I am using the model on (English, Chinese ...):

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [x] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [x] my own task or dataset: (give details below)

## To reproduce

(Pdb) net_inputs["attention_mask"]
tensor([[1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0],
        [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]], device='cuda:0')
(Pdb) net_inputs["lm_labels"]
tensor([[    3,  1489,    89,  -100,  -100,  -100],
        [ 5441,  1511,    13,     3, 10314,   152]], device='cuda:0')
(Pdb) net_inputs["lm_labels"]
tensor([[    3,  1489,    89,  -100,  -100,  -100],
        [ 5441,  1511,    13,     3, 10314,   152]], device='cuda:0')
(Pdb)

1.  model = T5ForConditionalGeneration.from_pretrained("t5-large")
2. outputs = model(input_ids=net_inputs["input_ids"], attention_mask=net_inputs["attention_mask"], lm_labels=net_inputs["lm_labels"])
3. loss, lm_logits = outputs[0], outputs[1]; print(loss)

## Expected behavior

Seems like a masking issue. 

## Environment info

     
- `transformers` version: 2.11.0
- Platform: V100
- Python version: 3.6.10
- PyTorch version (GPU?): 1.5.0
- Tensorflow version (GPU?):N/A
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: parallel

**Comment by patrickvonplaten** [comment]
See #4586.
